src/places/autocomplete_service.ts

```diff
@@ -182,23 +188,117 @@ export class MigrationAutocompleteService {
-    return new Promise((resolve) => {
-      this.getQueryPredictions(request, (predictions, status) => {
-        // Filter out predictions that don't have a place_id
-        const filteredPredictions = predictions.filter((prediction) => {
-          return prediction.place_id;
-        });
+  // https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompleteService.getPlacePredictions
+  getPlacePredictions(request, callback?): Promise<google.maps.places.AutocompleteResponse> {
+    return new Promise((resolve, reject) => {
+      const query = request.input;
+      const location = request.location; // optional
+      const locationBias = request.locationBias; // optional
+      const bounds = request.bounds || request.locationRestriction; // optional
+      const radius = request.radius; // optional
+      const language = request.language; // optional
 
-        // If a callback was given, invoke it before resolving the promise
-        if (callback) {
-          callback(filteredPredictions, status);
-        }
+      const input: AutocompleteRequest = {
+        QueryText: query, // required
+        MaxResults: 5, // Google only returns a max of 5 results
+        AdditionalFeatures: [SuggestAdditionalFeature.CORE], // Without this, only the ID and Title will be returned
+      };
+
+      // Set the appropriate location bias on our request input
+      setRequestLocationBias(input, location, locationBias, bounds, radius);
+
+      if (language) {
+        input.Language = language;
+      }
+
+      const command = new AutocompleteCommand(input);
+
+      this._client
+        .send(command)
+        .then((response) => {
+          const googlePredictions: google.maps.places.AutocompletePrediction[] = [];
+
+          const results = response.ResultItems;
+          if (results && results.length !== 0) {
+            results.forEach(function (result) {
+              const matchedSubstrings = [];
+              const addressHighlights = result.Highlights?.Address;
+              if (addressHighlights) {
+                Object.keys(addressHighlights).forEach((key) => {
+                  const highlights = addressHighlights[key];
+                  if (Array.isArray(highlights)) {
+                    highlights.forEach((highlight) => {
+                      matchedSubstrings.push({
+                        length: highlight.EndIndex - highlight.StartIndex,
+                        offset: highlight.StartIndex,
+                      });
+                    });
+                  }
+                });
+              }
 
-        resolve({
-          predictions: filteredPredictions,
+              const terms: google.maps.places.PredictionTerm[] = [];
+              const description = result.Address.Label;
+              let offset = 0;
+              for (let index = 0; index < description.length; index++) {
+                if (description[index] == ",") {
```

Comment:
Seems like this maybe be done more readable with string.split()?

Reply:
If we only needed the separate values, then yes, but we need to keep track of the `index` so that we can provide the `offset` for each term.